ledger-core/test/integration/CMakeLists.txt

```diff
@@ -14,27 +14,20 @@ add_library(ledger-core-integration-test
 
 target_include_directories(
     ledger-core-integration-test PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include/ledger-test>
+    # hack for centralized build
+    # including parent directory allows header inclusions such as "<integration/...>"
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}> $<INSTALL_INTERFACE:include/ledger-test>
 )
 
 target_link_libraries(ledger-core-integration-test gtest gtest_main)
 
 if(SYS_OPENSSL)
-    find_package(OpenSSL)
-    target_link_libraries(ledger-core-integration-test OpenSSL::crypto)
+    find_package(OpenSSL REQUIRED COMPONENTS Crypto)
+    target_link_libraries(ledger-core-integration-test OpenSSL::Crypto)
 else()
     target_link_libraries(ledger-core-integration-test crypto)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    # We need this otherwise link fails (for Linux),
-    # this is needed since we build static library for tests,
-    # not needed if we link against dynamic
-    # TODO: needs more investigation
-    target_link_libraries(ledger-core-integration-test ledger-core)
-else()
-    target_link_libraries(ledger-core-integration-test ledger-core-static)
-endif()
-
+target_link_libraries(ledger-core-integration-test ledger-core-bundle)
```

Comment:
This is really weird to link the `root` target here. I think a better option is to link `ledger-core-api` or `ledger-core-obj`, right ?

Reply:
No, because you need the actual objects. If you link `ledger-core-api`, you get only the interface. 

Also, integration testing should test the product / library, not sub-parts. It should test what the clients will use.